File: prisma/refs/py/set-matrix-zeroes.py
def set_zeroes(matrix):
    rows = {r for r, row in enumerate(matrix) for v in row if v == 0}
    cols = {c for row in matrix for c, v in enumerate(row) if v == 0}
    for r, row in enumerate(matrix):
        for c in range(len(row)):
            if r in rows or c in cols:
                row[c] = 0

File: prisma/refs/java/set-matrix-zeroes.java
import java.util.*;

class Solution {
    public void setZeroes(int[][] matrix) {
        int m = matrix.length, n = matrix[0].length;
        Set<Integer> rows = new HashSet<>(), cols = new HashSet<>();
        for (int r = 0; r < m; r++)
            for (int c = 0; c < n; c++)
                if (matrix[r][c] == 0) { rows.add(r); cols.add(c); }
        for (int r = 0; r < m; r++)
            for (int c = 0; c < n; c++)
                if (rows.contains(r) || cols.contains(c)) matrix[r][c] = 0;
    }
}
